### PR TITLE
fix: update IG validation to reject reference params with no target

### PR DESCRIPTION
**Pull request by ssvegaraju**
Issue #, if available:
[#449](https://github.com/awslabs/fhir-works-on-aws-deployment/issues/449)
Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/implementationGuides/index.test.ts [hunk 1]
```diff
@@ -220,4 +221,20 @@ describe('compile', () => {
         ]);
         await expect(compiled).rejects.toThrowError();
     });
+    test(`search param of type reference with no target`, async () => {
```
Comment (carvantes) [inline src/implementationGuides/index.test.ts:224]: Add a test for target being an empty array. I think a code change is needed to handle that case properly

**Review by carvantes — CHANGES_REQUESTED** [review]

**Comment by carvantes** [comment]
Also, the error msg of `Error: The following input is not a search parameter: {...}` is not that helpful. The `isFhirSearchParam` is a very basic check to make sure that required fields are present and that the field types are correct. 

Maybe we can create a separate method for semantic validations like this that throws more specific error messages.

File: src/implementationGuides/index.test.ts (hunk 1 above)
Comment (ssvegaraju) [reply src/implementationGuides/index.test.ts:224]: Good catch, I added a test case for this!

**Review by carvantes — APPROVED** [review]
